### PR TITLE
Any stat

### PR DESCRIPTION
**Pull request by sundevour**
this streamlines the stat comparison function and undoes the if-else nightmare i created

then it turns stat comparison function into a multiple stat comparer that checks a list of stats against the value when "any" is passed as the stat.
this allows users to check for desirable things on armor, like "any armor stat over 25" or my favorite, "any stat is the highest among available armor pieces"

### CODE REVIEW

File: src/app/search/search-filter-hashes.ts [hunk 1]
```diff
@@ -199,16 +199,30 @@ export const emptySocketHashes = [
...
+const realArmorStatHashByName = {
   mobility: 2996146975,
   resilience: 392767087,
   recovery: 1943323491,
   discipline: 1735777505,
   intellect: 144602215,
-  strength: 4244567218,
+  strength: 4244567218
+};
+/** these stats actually exist on DIM armor */
```
Comment (bhollis) [inline src/app/search/search-filter-hashes.ts]: ???

Comment (sundevour) [reply src/app/search/search-filter-hashes.ts]: i'll clarify a bit more in comments. there's different groupings of stats based on what we want to do with them

"any" doesn't exist at all and is just included in filter name creation loops, total is worth including in the list for collecting max stats and does belong in a `name->hash` translator table. will rearrange a bit.

Comment (bhollis) [reply src/app/search/search-filter-hashes.ts]: I see, I got confused by the distinction between "these exist on D2 armor" vs. "these exist on DIM armor".

**Comment by sundevour** [comment]
merging this with a missing i18n string then fixing, because its outdated and keeps deleting a bunch of stuff when i `yarn i18n`

**Pull request merged**